litmus-portal/frontend/src/components/CheckBox/index.tsx

```diff
@@ -1,9 +1,29 @@
 import { Checkbox, withStyles } from '@material-ui/core';
 
-const CheckBox = withStyles((theme) => ({
+export const CheckBoxWhite = withStyles((theme) => ({
   root: {
-    color: theme.palette.text.hint,
+    color: theme.palette.border.main,
+    '&$checked': {
+      color: theme.palette.background.paper,
+      '& .MuiSvgIcon-root': {
+        background: theme.palette.secondary.main,
+        borderRadius: '0.25rem',
+      },
+    },
   },
+  checked: {},
 }))(Checkbox);
 
-export default CheckBox;
+export const CheckBoxBlue = withStyles((theme) => ({
+  root: {
+    color: theme.palette.border.main,
+    '&$checked': {
+      color: theme.palette.cards.header,
+      '& .MuiSvgIcon-root': {
+        background: theme.palette.secondary.main,
+        borderRadius: '0.25rem',
+      },
+    },
+  },
+  checked: {},
```

Comment:
same

Reply:
same as above

Reply:
Do we require to have two components `CheckBoxBlue` and `CheckBoxWhite`? May we please have only one component and override the styles for the other. Or we don't create the local component and simply pass the styles as required.

Reply:
Yes it should be from MUI for now we can use local one, also the two components were required here because of different colour properties which cannot be passed on class and have to be overridden on MUI component. (when using MUI we can shift to using a single one with proper style interface, workflow analytics team is already working on the same) this PR is just for fixing the colour.

Reply:
try passing the color as a prop to your component....withStyles is a HOC it should have an API to pass props @ishangupta-ds

Reply:
Using the checkbox compoent in (https://github.com/whitetiger1399/litmus-ui/tree/autoCompleteChip).  
![Screenshot from 2021-05-12 18-10-10](https://user-images.githubusercontent.com/18084015/117976537-86c76400-b34d-11eb-8123-b118bf9e2876.png)
![Screenshot from 2021-05-12 18-10-22](https://user-images.githubusercontent.com/18084015/117976547-88912780-b34d-11eb-9fe2-921e648ac3aa.png) 
We may pass a className which will override the default m-ui properties as per your requirement

Reply:
Tried out the above approach @whitetiger1399 , it does not work for the selectors for base class properties its working fine like in above screenshot as they are styles for root `checked` and `MuiSvgIcon-root` are not getting overridden, so only colour worked from the class mentioned below but we need both of them as an override. Please try to check if these are working for you in litmus-ui for select chip.

  rootCheck: {
    color: theme.palette.border.main,
    '&$checked': {
      color: theme.palette.background.paper,
      '& .MuiSvgIcon-root': {
        background: theme.palette.secondary.main,
        borderRadius: '0.25rem',
      },
    },
  },
  checkedCheck: {},
  
  
<img width="1023" alt="Screenshot 2021-05-12 at 7 40 18 PM" src="https://user-images.githubusercontent.com/50652677/117990362-bc267e80-b35a-11eb-863c-c7b22779eebe.png">

Reply:
@arkajyotiMukherjee makeStyle allows props directly, we can use useStyles but then it no longer stays a pure react component so it cannot be used in jsx directly unless we extend the Checkbox from MUI for this its better to do it in litmus-ui than in litmus.

useStyles:

export const StyleWithThemeProps = (props: any) => {
  return withStyles(
    (theme) => ({
      root: {
        color: theme.palette.border.main,
        '&$checked': {
          color: props.checkedColor,
          '& .MuiSvgIcon-root': {
            background: theme.palette.secondary.main,
            borderRadius: '0.25rem',
          },
        },
      },
      checked: {},
    }),
    { withTheme: true }
  )(Checkbox);
};

makeStyles example:

const useStyles = makeStyles({
  root: {
    background: props => props.color,
    "&:hover": {
      background: props => props.hover
    },
    border: 0,
    borderRadius: 3,
    color: "white",
    height: 48,
    padding: "0 30px"
  }
});

export default function Hook() {
  const classes = useStyles({
    color: "red",
    hover: "blue"
  });
  return <Button className={classes.root}>Hook</Button>;
}

Reply:
For now till we use from litmus-ui we may need to use these two styled/overrided components separately because MUI is not offering a direct way to override the css selectors without overriding the base styles by wrapping it up with the withStyles HOC @arkajyotiMukherjee @whitetiger1399

Reply:
Ignore the  comments @arkajyotiMukherjee and @whitetiger1399 the props method worked !

Reply:
Although it worked we still have the same issue that the overriding failed after using props.

<img width="356" alt="Screenshot 2021-05-13 at 8 57 01 AM" src="https://user-images.githubusercontent.com/50652677/118073316-3857ab00-b3c9-11eb-9cc1-e90fd8833c38.png">